### ISSUE TITLE
BUG: Polynomial degree doesn't account for 0 coefficient

### ISSUE THREAD

**Issue by stefmolin** [numpy/numpy#23531]
### Describe the issue:

I was under the assumption that the `Polynomial.degree()` method yielded the degree of the polynomial, but after looking at the code in main and the return value in the docs, it just subtracts one from the number of coefficients, which doesn't account for a 0 in the highest degree coefficient (possible if another process determines the coefficients that get passed in).

### Reproduce the code example:

```python
import numpy as np
np.polynomial.Polynomial([1, 7, 4, 0]).degree()  # returns 3, but I would expect 2
```

### Runtime information:

1.23.1
3.8.11 (default, Jul 29 2021, 14:57:32)
[Clang 12.0.0 ]

### Context for the issue:

I'm happy to put in a PR if this is not by design.

**Comment by charris** [comment]
There is a `trim` method for that. Zero in the numerical sense can be a little fuzzy, so removing it is left to the user's discretion.

**Comment by stefmolin** [comment]
Makes sense.

**Comment by stefmolin** [comment]
Agreed. Where would be the best spot for the example? The class itself, the `degree()` method, and/or the `trim()` method?

**Comment by rossbar** [comment]
My vote would be the examples section for the `degree` method docstring - perhaps it could be added to #23530
